### PR TITLE
TOAZ-242: update sam resource type to kubernetes-app

### PR DESCRIPTION
**Pull request by rtitle**
Depends on https://github.com/broadinstitute/cromwhelm/pull/110 and https://github.com/DataBiosphere/terra-azure-relay-listeners/pull/22

Tested on a BEE.

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green (you can re-run automation tests with a comment saying `jenkins retest`
- [ ] Run the automation tests multiple times in parallel to weed out instability if applicable via a comment saying `multi-test`
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -29,8 +29,8 @@ ENV TERRA_APP_VERSION 0.5.0
 ENV GALAXY_VERSION 1.6.1
 ENV NGINX_VERSION 4.3.0
 # If you update this here, make sure to also update reference.conf:
-ENV CROMWELL_CHART_VERSION 0.2.154
-ENV CROWELL_ON_AZURE_CHART_VERSION 0.2.154
+ENV CROMWELL_CHART_VERSION 0.2.158
```
Comment (rtitle) [inline Dockerfile:32]: There are quite a few places to update for a chart version bump.. 🤔

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala [hunk 2]
```diff
@@ -198,8 +198,8 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
         raw"relaylistener.targetHost=http://coa-${release.asString}-reverse-proxy-service:8000/",
         raw"relaylistener.samUrl=${config.samConfig.server}",
         raw"relaylistener.samResourceId=${samResourceId.resourceId}",
-        // TODO (TOAZ-242): change to kubernetes-app resource type once listener supports it
-        raw"relaylistener.samResourceType=controlled-application-private-workspace-resource",
+        raw"relaylistener.samResourceType=kubernetes-app",
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala:247]: This is the intended resource type -- the relay listener now supports it :)

**Comment by rtitle** [comment]
fiab-start failure, jenkins retest

**Review by Ghost-in-a-Jar — APPROVED** [review]

**Comment by rtitle** [comment]
fiab-start failed again, looks like Argo timeout

jenkins retest

**Pull request merged**